pypsa/components.py

```diff
@@ -1911,29 +1895,19 @@ def get_bounds_pu(
             min_pu = pd.DataFrame(0, index=max_pu.index, columns=max_pu.columns)
             if attr == "p_store":
                 max_pu = -self.get_switchable_as_dense(c, min_pu_str, sns, inds=index)
-            if (
-                attr == "state_of_charge"
-            ):  # this is a special case. If it can be done simpler, please do it
-                max_hours = self.df(c).max_hours.T
-                _sns = self.snapshots.get_level_values("snapshot").unique()
-                new_index = pd.MultiIndex.from_product(
-                    [max_hours.index, _sns], names=["scenario", "snapshot"]
-                )
-                max_pu = max_hours.loc[max_hours.index.repeat(len(_sns))].set_index(
-                    new_index
-                )
-                min_pu = pd.DataFrame(0, *max_pu.axes)
+            if attr == "state_of_charge":
+                max_pu = self.get_switchable_as_dense(c, "max_hours", sns, inds=index)
```

Comment:
@FabianHofmann for `state_of_charge` case this line does not work since
`sn.pnl(component).keys()` does not contain `max_hours` attr.
we can (i) roll back to my workaround, or (ii) more generally, make `get_stichable_as_dense` handle such cases. 
go for the latter?

Reply:
yes, let's go for the latter. I'll push a commit